File: docker/kernel/project/ViajesAltairis.Domain/Entities/Provider.cs
namespace ViajesAltairis.Domain.Entities;

public class Provider : BaseEntity
{
    public long TypeId { get; set; }
    public long CurrencyId { get; set; }
    public string Name { get; set; } = null!;
    public string? ApiUrl { get; set; }
    public string? ApiUsername { get; set; }
    public string? ApiPasswordEncrypted { get; set; }
    public decimal Margin { get; set; }
    public bool Enabled { get; set; }
    public string? SyncStatus { get; set; }
    public DateTime? LastSyncedAt { get; set; }

    public ProviderType Type { get; set; } = null!;
    public ICollection<HotelProvider> HotelProviders { get; set; } = [];
    public ICollection<User> Users { get; set; } = [];
}

File: docker/admin-web/project/src/types/provider.ts
export interface ProviderDto {
  id: number;
  typeId: number;
  currencyId: number;
  name: string;
  apiUrl: string | null;
  apiUsername: string | null;
  margin: number;
  enabled: boolean;
  syncStatus: string | null;
  lastSyncedAt: string | null;
  createdAt: string;
}

export interface CreateProviderRequest {
  typeId: number;
  currencyId: number;
  name: string;
  apiUrl: string | null;
  apiUsername: string | null;
  apiPassword: string | null;
  margin: number;
}

export interface ProviderTypeDto {
  id: number;
  name: string;
  createdAt: string;
}
